### PR TITLE
Rewrote disk_info to use Windows API calls instead of WMIC. Renamed o…

### LINKED ISSUE #93: Winfetch 2.1.0 taking ages to get disk info
Following the release of winfetch 2.0.0 or something close to that, I have noticed that winfetch hangs somewhere between printing RAM info and pritnign disk info. I usually Ctrl+C it after about 5 seconds, but I left it running once and PS reported roughly 50 seconds to get through it.

I am using PowerShell 7, winfetch 2.1.0, Windows 10, disk is 1 TB WD Blue SSD.

### PR DESCRIPTION
**Pull request by AwesomeCronk**
…ld disk_info to disk_info_CIM.

Fixes #93

### CODE REVIEW

**Comment by rashil2000** [comment]
I'm only getting C: drive's information when ShowDisks is set to *.

It showed all three (C:, D:, E:) before.

**Comment by AwesomeCronk** [comment]
Interesting… Not sure what caused that because I was able to get an array of all the disks in my system, even my USB hub’s attachment points. I must have forgotten to actually show all disks and figures only C: chowed up because it was the only one with media in it.

**Comment by rashil2000** [comment]
Hmm. All 3 of my drives have significant usage so this is probably something that we'll have to fix before this is merged.

**Comment by AwesomeCronk** [comment]
Yes, it most definitely is. I am in the middle of some sleuthing around and I have found that my method of checking if the array `$showDisks` contains each `$diskLetter` was not working. I redid that and got that part to work, but now I have come across the issue that from some reason, the values that `[WinAPI.DiskMethods]::GetDiskFreeSpaceEx` writes to are overridden to `0` on each consecutive pass. `C:` is my only drive that isn't virtual or an empty hub/cd drive, so I am  going to do some testing with media in them.

**Comment by AwesomeCronk** [comment]
Decided to check the success flag returned by the API call. Gotta love this...
image

**Comment by AwesomeCronk** [comment]
Running the same loop that checks the disk usage in winfetch at the terminal works, though it fails inside winfetch. No idea why.

**Comment by AwesomeCronk** [comment]
Much better!
image
I found it to be related to the PowerShell/C# interop layer. For whatever reason, the `System.String[]` that splitting the C# `string[]` returned for `GetLogicalDriveStringsW` was designed to be close enough to `System.Object[]` that I could do all that without errors, but refused to pass correctly to `GetDiskFreeSpaceEx`. That should all be fixed now.

**Comment by rashil2000** [comment]
Great! It's working slick now.

If you have a Win 7 machine or VM lying around, please test it there. If there are no problems, we can get rid of the `info_disk_CIM` function.

**Comment by AwesomeCronk** [comment]
I lack a win7 machine. I’ll ask around and see if I know someone who does and if I can find someone I’ll see if they are ok with testing it.

**Comment by rashil2000** [comment]
Just tested it in a VM, works fine.

Please remove the old function.

**Comment by AwesomeCronk** [comment]
Will do 👍

File: winfetch.ps1 [hunk 1]
```diff
@@ -671,29 +709,52 @@ function info_disk {
...
+                title   = "Disk ($diskLetter)"
+                content = get_level_info "" $diskstyle $usage "$(to_units $used) / $(to_units $total)"
+            })
+        }
+    }
+
     return $lines
 }
 
-
```
Comment (rashil2000) [inline winfetch.ps1:696]: Let's keep this line, we have a gap of 2 newlines between each top level function.

**Review by rashil2000 — APPROVED** [review]

**Comment by rashil2000** [comment]
Great, thanks for the awesome contribution!

**Review by jcwillox — APPROVED** [review]
Great job, seems to work perfectly and is even faster than before 👍

**Comment by AwesomeCronk** [comment]
Thank you! Happy to have been able to help!

**Pull request merged**